examples/sysctrl/hello-sysctrl/examples/init_hpc.rs

```diff
@@ -99,22 +100,43 @@ fn main() -> ! {
 
     print_example_name!();
 
+    // Enable SDRAM
+    let ddr_mode = 0x1;
+    let axi_enable = 0x2;
+    sprint!("Enabling SDRAM ddr_mode={:#x}, axi_enable={:#x}...",
+            ddr_mode,
+            axi_enable,
+    );
+    soc_ctrl::sdram_cfg_axi_ddr_mode_mask(ddr_mode);
+    soc_ctrl::sdram_cfg_axi_enable_mask(axi_enable);
+    sprintln!(" done");
+
     for i in 0..5 {
         let addr = HPC_BOOTRAM_ADDR + i * 4;
         sprint!("Writing 0x6f into {:#x}...", addr);
         unsafe { ptr::write_volatile(addr as *mut _, 0x6f) };
         sprintln!(" done");
     }
 
-    let hpc_core_en = 0xf;
+    // Configure execute regions for SDRAM
+    let execute_region_pattern = 0x6fff_ffff;
```

Comment:
I think this should be `0x7000_0000`. The HPC documentation is distracting a little bit here. I don't think there's any reason why the last address of SDRAM should be inaccessible to HPC.

Reply:
We should also specify `executeregionlength3=0x2000_0000 `, `executeregionlength3=0x10_0000`. 

Ref: https://soc-hub.gitlab-pages.tuni.fi/headsail/hw/hpc-ss/hpcss_documentation.html#address-map

Reply:
This most likely relates to a bug on Ballast: reported at:

- https://gitlab.tuni.fi/soc-hub/ballast/ballast/-/issues/2
- https://gitlab.tuni.fi/soc-hub/ballast/ballast/-/blob/master/ERRATA.md?ref_type=heads#last-byte-of-some-regions-is-inaccessible

Reply:
Based on this: https://gitlab.tuni.fi/soc-hub/headsail/hw/interconnect-ss/-/blob/main/rtl/interconnect_core.v

The bug is no longer present on Headsail for any of the regions. Therefore we should use the entire range (last address included).

Reply:
Execute region extended to the full width of the SDRAM. Looks like this at least partially fixed the issue related to running code from SDRAM as the `apb_uart0` now runs without the breakpoint.

Added execution region configuration for SRAM and C2C as suggested.

Reply:
Previous comment about SDRAM working was wrong, turns our I was running from bootram